ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala

```diff
@@ -173,3 +173,49 @@ final class CommandsValidator(ledgerId: LedgerId) {
         Left(missingField("command"))
     }
 }
+
+object CommandsValidator {
+
+  case class Submitters[T](actAs: Set[T], readAs: Set[T])
```

Comment:
Maybe at least worth adding a comment that `actAs` must be non-empty if you don’t want to enforce it in the type?